app/src/hid.c

```diff
@@ -55,7 +55,9 @@ int zmk_hid_unregister_mod(zmk_mod_t modifier) {
             continue;                                                                              \
         }                                                                                          \
         keyboard_report.body.keys[idx] = val;                                                      \
-        break;                                                                                     \
+        if (val) {                                                                                 \
+            break;                                                                                 \
+        }                                                                                          \
```

Comment:
I think I appreciate what this does, but what precipitated it?  Please could you add the reasoning to the commit body.

Reply:
Added a bit more detail to the commit message.

Reply:
ah, so it's the discards that leads to this scenario - yep, makes sense.  Thanks!